packages/block-library/src/embed/edit.js

```diff
@@ -0,0 +1,382 @@
+					const potentialRatio = ASPECT_RATIOS[ ratioIndex ];
+					if ( aspectRatio >= potentialRatio.ratio ) {
+						return {
+							[ potentialRatio.className ]: allowResponsive,
+							'wp-has-aspect-ratio': allowResponsive,
+						};
+					}
+				}
+			}
+
+			return this.props.attributes.className;
+		}
+
+		/**
+		 * Sets the aspect ratio related class names returned by `getAspectRatioClassNames`
+		 * if `allowResponsive` is truthy.
+		 *
+		 * @param {string} html The preview HTML.
+		 */
+		setAspectRatioClassNames( html ) {
+			const { allowResponsive } = this.props.attributes;
+			if ( ! allowResponsive ) {
+				return;
+			}
+			const className = classnames(
+				this.props.attributes.className,
+				this.getAspectRatioClassNames( html )
+			);
+			this.props.setAttributes( { className } );
+		}
+
+		/***
+		 * Gets block attributes based on the preview and responsive state.
+		 *
+		 * @param {string} preview The preview data.
+		 * @param {boolean} allowResponsive Apply responsive classes to fixed size content.
+		 * @return {Object} Attributes and values.
+		 */
+		getAttributesFromPreview( preview, allowResponsive = true ) {
+			const attributes = {};
+			// Some plugins only return HTML with no type info, so default this to 'rich'.
+			let { type = 'rich' } = preview;
+			// If we got a provider name from the API, use it for the slug, otherwise we use the title,
+			// because not all embed code gives us a provider name.
+			const { html, provider_name: providerName } = preview;
+			const providerNameSlug = kebabCase( toLower( '' !== providerName ? providerName : title ) );
+
+			if ( isFromWordPress( html ) ) {
+				type = 'wp-embed';
+			}
+
+			if ( html || 'photo' === type ) {
+				attributes.type = type;
+				attributes.providerNameSlug = providerNameSlug;
+			}
+
+			attributes.className = classnames(
+				this.props.attributes.className,
+				this.getAspectRatioClassNames( html, allowResponsive )
+			);
+
+			return attributes;
+		}
+
+		/***
+		 * Sets block attributes based on the preview data.
+		 */
+		setAttributesFromPreview() {
+			const { setAttributes, preview } = this.props;
+			const { allowResponsive } = this.props.attributes;
+			setAttributes( this.getAttributesFromPreview( preview, allowResponsive ) );
+		}
+
+		switchBackToURLInput() {
+			this.setState( { editingURL: true } );
+		}
+
+		getResponsiveHelp( checked ) {
+			return checked ? __( 'Videos and other content automatically resizes.' ) : __( 'Content is fixed size.' );
+		}
+
+		toggleResponsive() {
+			const { allowResponsive, className } = this.props.attributes;
+			const { html } = this.props.preview;
+			const responsiveClassNames = this.getAspectRatioClassNames( html, ! allowResponsive );
+
+			this.props.setAttributes(
+				{
+					allowResponsive: ! allowResponsive,
+					className: classnames( className, responsiveClassNames ),
+				}
+			);
+		}
+
+		render() {
```

Comment:
The render function is pretty massive and could be broken up into separate components